src/components/Letter.tsx

```diff
@@ -0,0 +1,287 @@
+import {
+  LetterInterface,
+  LetterPersistenceData,
+  LetterSharedData,
+  LetterType,
+} from "../types";
+import { motion } from "framer-motion";
+import { useRef, useState } from "react";
+import Draggable from "react-draggable";
+import dayjs from "dayjs";
+import Y from "yjs";
+
+interface Props {
+  letter: LetterInterface;
+  shared?: Y.Map<LetterSharedData>;
+  isEditable?: boolean;
+  disableDrag?: boolean;
+}
+
+export function Letter({ letter, shared, isEditable, disableDrag }: Props) {
+  const [isDragging, setDragging] = useState(disableDrag ? true : false);
+  const { id, initialPersistenceData } = letter;
+  const saved = disableDrag ? undefined : localStorage.getItem(id);
+  const savedPersistenceData = useRef<LetterPersistenceData>(
+    saved ? JSON.parse(saved) : {}
+  );
+  const position = {
+    ...initialPersistenceData,
+    ...savedPersistenceData.current,
+  };
+  // TODO: annotate this with the total number persisted in server.
+  const currentSharedData = shared?.get(id) || {
+    numOpens: 0,
+    numDrags: 0,
+  };
+
+  const letterContent = (
+    <div>
+      <motion.div
+        className="letter"
+        transition={{
+          type: "spring",
+          stiffness: 150,
+          duration: 0.3,
+          bounce: 0.8,
+        }}
+        // TODO: move rotation to css random generation
+        animate={{
+          transform: position
+            ? `rotate(${isDragging ? 0 : position.rotation}deg)`
+            : "",
+          boxShadow: isDragging
+            ? "0 0 35px rgba(51, 75, 97, 0.35)"
+            : "0 0 5px rgba(51, 75, 97, 0.2)",
+        }}
+        initial={false}
+      >
+        <LetterView
+          letter={letter}
+          shared={shared}
+          isDragging={isDragging}
+          isEditable={isEditable}
+        />
+      </motion.div>
+    </div>
+  );
+
+  return disableDrag ? (
+    <div className="letter-container disabled">{letterContent}</div>
+  ) : (
+    <Draggable
+      handle=".letterHead"
+      defaultClassName="letter-container"
+      defaultPosition={position}
+      onStart={() => {
+        setDragging(true);
+        // TODO: update z-index to be the highest.
```

Comment:
I tried that with the CSS in `animate` on the `motion.div` but it didn't seem to work, I'll poke more into it later